INSTALL.md

```diff
@@ -5,6 +5,8 @@ System requirements: GalSim currently only supports Linux and Mac OSX.
 
 Table of Contents:
 
+0. [Overall summary](#0-overall-summary)
```

Comment:
Thanks for listing all those alternatives!  Since the section headings themselves are fine, I went the lazy route of just converting to ```0)``` and so on.  Now the list itself looks fine to me in markdown format.   Possibly this is a bit weird but I wanted to indicate it's a summary before all the detailed bits, so unless you object, let's just go with this...